apps/gitness/src/pages/project-settings/project-settings-new-member-page.tsx

```diff
@@ -0,0 +1,214 @@
+      },
+      onError: error => {
+        alert('Error adding member: ' + error.message)
+      }
+    }
+  )
+
+  // Form submit handler
+  const onSubmit: SubmitHandler<NewMemberFields> = data => {
+    addMember({ body: { user_uid: data.memberName, role: data.role as EnumMembershipRole } })
+  }
+
+  // Handle member selection
+  const handleMemberSelect = (uid: string, display_name: string) => {
+    setSelectedMember(display_name)
+    setValue('memberName', uid, { shouldValidate: true })
+  }
+
+  const handleCancel = () => {
+    resetNewMemberForm()
+    navigate(`/spaces/${space_ref}/settings/members`, { replace: true })
+  }
+
+  const newMemberRoleValue = watch('role')
+
+  return (
+    <SandboxLayout.Main hasLeftPanel hasHeader hasSubHeader>
+      <SandboxLayout.Content maxWidth="2xl">
+        <Spacer size={10} />
+        <Text size={5} weight="medium">
+          Add a new member to {space_ref}
+        </Text>
+        <Spacer size={6} />
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <FormFieldSet.Root>
+            {/* New Member Name */}
+            <FormFieldSet.ControlGroup>
+              <FormFieldSet.Label htmlFor="memberName" required>
+                New Member Name
+              </FormFieldSet.Label>
+              <DropdownMenu>
+                <DropdownMenuTrigger>
+                  <div className="flex justify-between border rounded-md items-center">
+                    <Button variant="ghost">
+                      <Text>{selectedMember || 'Select New Member'}</Text>
+                    </Button>
+                  </div>
+                </DropdownMenuTrigger>
+                <DropdownMenuContent style={{ width: 'var(--radix-dropdown-menu-trigger-width)' }}>
+                  <DropdownMenuLabel>
+                    {usersData?.length !== 0 ? (
+                      'Available users'
+                    ) : (
+                      <Text color="tertiaryBackground">No available users</Text>
+                    )}
+                  </DropdownMenuLabel>
+                  {usersData && <DropdownMenuSeparator />}
+                  {usersData &&
+                    usersData.map(user => (
+                      <DropdownMenuItem
+                        key={user.uid}
+                        onSelect={() => handleMemberSelect(user.uid ?? '', user.display_name ?? '')}>
+                        {user.display_name}
+                      </DropdownMenuItem>
+                    ))}
+                </DropdownMenuContent>
+              </DropdownMenu>
+              {/* Register the field for validation */}
+              <input type="hidden" {...register('memberName', { required: 'Please select a member' })} />
+              {errors.memberName && (
+                <FormFieldSet.Message theme={FormFieldSet.MessageTheme.ERROR}>
+                  {errors.memberName.message}
+                </FormFieldSet.Message>
+              )}
```

Comment:
All of this can be replaced with one `<FormMessage />`, and it should show up correctly when there's an error

Reply:
it will show getFieldState error if the value is empty. I will check on that in follow up PR